resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/K8sSubmitOps.scala

```diff
@@ -0,0 +1,180 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.k8s.submit
+
+import scala.collection.JavaConverters._
+
+import io.fabric8.kubernetes.api.model.Pod
+import io.fabric8.kubernetes.client.KubernetesClient
+
+import org.apache.spark.SparkConf
+import org.apache.spark.deploy.{SparkSubmitArguments, SparkSubmitOperationsClient}
+import org.apache.spark.deploy.k8s.{KubernetesUtils, SparkKubernetesClientFactory}
+import org.apache.spark.deploy.k8s.Config.{KUBERNETES_AUTH_SUBMISSION_CONF_PREFIX, KUBERNETES_SUBMIT_GRACE_PERIOD}
+import org.apache.spark.deploy.k8s.Constants.{SPARK_POD_DRIVER_ROLE, SPARK_ROLE_LABEL}
+import org.apache.spark.deploy.k8s.KubernetesUtils.formatPodState
+import org.apache.spark.util.{CommandLineLoggingUtils, Utils}
+
+private [spark] sealed trait K8sSubmitOp extends CommandLineLoggingUtils {
+  def executeOnPod(pName: String, ns: String, gracePeriodSeconds: Option[Long])
+      (implicit client: KubernetesClient): Unit
+  def executeOnGlob(pods: List[Pod], ns: String, gracePeriodSeconds: Option[Long])
+      (implicit client: KubernetesClient): Unit
+}
+
+private [spark] class KillApplication extends K8sSubmitOp  {
+  override def executeOnPod(pName: String, ns: String, gracePeriodSeconds: Option[Long])
+      (implicit client: KubernetesClient): Unit = {
+    val podToDelete = client
+      .pods
+      .inNamespace(ns)
+      .withName(pName)
+      gracePeriodSeconds match {
+        case Some(period) => podToDelete.withGracePeriod(period).delete()
+        case _ => podToDelete.delete()
+      }
+  }
+
+  override def executeOnGlob(pods: List[Pod], ns: String, gracePeriodSeconds: Option[Long])
+      (implicit client: KubernetesClient): Unit = {
+    if (pods.nonEmpty) {
+      pods.foreach(pod => printMessage(s"Deleting driver pod: ${pod.getMetadata.getName}."))
+      val nClient = client
+        .pods
+        .inNamespace(ns)
+      gracePeriodSeconds match {
+        case Some(period) =>
+          // this is not using the batch api because no option is provided
+          // when using the grace period.
+          pods.foreach { pod =>
+             nClient
+              .withName(pod.getMetadata.getName)
+               .withGracePeriod(period)
+              .delete()
+          }
+        case _ => nClient.delete(pods.asJava)
+      }
+    } else {
+      printMessage("No applications found.")
+    }
+  }
+}
+
+private [spark] class ListStatus extends K8sSubmitOp {
+  override def executeOnPod(pName: String, ns: String, gracePerio: Option[Long])
+      (implicit client: KubernetesClient): Unit = {
+      val pod = client
```

Comment:
indentation is wrong

Reply:
The implicit part should have 2 spaces or 4? Currently im using 4.

Reply:
arguments = 4, code = 2.